src/Std/Data/DHashMap/Internal/RawLemmas.lean

```diff
@@ -2684,6 +2685,279 @@ end Const
+    (h₂ : m₂.val.WF) {k fallback : α} (h' : m₂.contains k = false) :
+    (m₁.union m₂).getKeyD k fallback = m₁.getKeyD k fallback := by
+  revert h'
+  simp_to_model [contains, union, getKeyD]
+  intro h'
+  apply getKeyD_insertList_of_contains_right_eq_false h'
+
+theorem getKeyD_union_of_mem_right [EquivBEq α] [LawfulHashable α] (h₁ : m₁.val.WF)
+    (h₂ : m₂.val.WF) {k fallback : α} (mem : m₂.contains k) :
+    (m₁.union m₂).getKeyD k fallback = m₂.getKeyD k fallback := by
+  simp_to_model [contains, union, getKeyD] using List.getKeyD_insertList_of_mem_right
+  revert mem
+  simp_to_model [contains]
+  simp only [imp_self]
+
+theorem getKeyD_union_of_mem_right_lawful [LawfulBEq α] [LawfulHashable α] (h₁ : m₁.val.WF)
+    (h₂ : m₂.val.WF) {k fallback : α} (mem : m₂.contains k) :
+    (m₁.union m₂).getKeyD k fallback = k := by
+  simp_to_model [contains, union, getKeyD] using List.getKeyD_insertList_of_mem_right_lawful
+  revert mem
+  simp_to_model [contains]
+  simp only [imp_self]
+
+theorem get?_union_of_contains_left_eq_false [LawfulBEq α] (h₁ : m₁.val.WF) (h₂ : m₂.val.WF)
+    {k : α} (contains_eq_false : m₁.contains k = false) :
+    (m₁.union m₂).get? k = m₂.get? k := by
+  revert contains_eq_false
+  simp_to_model [union, contains, get?] using getValueCast?_insertList_of_contains_left_eq_false
+
+theorem get_union_of_contains_left_eq_false [LawfulBEq α] (h₁ : m₁.val.WF) (h₂ : m₂.val.WF)
+    {k : α} (contains_eq_false : m₁.contains k = false) {h'} :
+    (m₁.union m₂).get k h' = m₂.get k (contains_of_contains_union_of_contains_left_eq_false h₁ h₂ h' contains_eq_false) := by
+  revert contains_eq_false
+  simp_to_model [union, contains, get] using List.getValueCast_insertList_of_contains_left_eq_false
+
+theorem getKey?_union_of_contains_left_of_contains_right_eq_false [EquivBEq α] [LawfulHashable α]
+    (h₁ : m₁.val.WF) (h₂ : m₂.val.WF)
+    {k : α} (mem : m₁.contains k) (not_mem : m₂.contains k = false) :
+    (m₁.union m₂).getKey? k = m₁.getKey? k := by
+  revert mem not_mem
+  simp_to_model [contains, getKey?, union]
+  intro mem not_mem
+  apply List.getKey?_insertList_of_contains_left_of_contains_right_eq_false
+  . exact not_mem
+
+theorem getKey?_union_of_contains_left_of_contains_right_eq_false_lawful [LawfulBEq α] [LawfulHashable α]
+    (h₁ : m₁.val.WF) (h₂ : m₂.val.WF)
+    {k : α} (mem : m₁.contains k) (not_mem : m₂.contains k = false) :
+    (m₁.union m₂).getKey? k = some k := by
+  revert mem not_mem
+  simp_to_model [contains, getKey?, union]
+  intro not_mem mem
+  apply List.getKey?_insertList_of_mem_of_not_mem
+  . exact not_mem
+  . exact mem
+
+theorem getKey_union_of_contains_left_of_contains_right_eq_false
+    [EquivBEq α] [LawfulHashable α] (h₁ : m₁.val.WF) (h₂ : m₂.val.WF)
+    {k : α} (mem : m₁.contains k) :
+    (∃ (mem₂ : m₂.contains k), (m₁.union m₂).getKey k (contains_union_of_left h₁ h₂ mem) = m₂.getKey k mem₂)
+    ∨ (m₁.union m₂).getKey k (contains_union_of_left h₁ h₂ mem) = m₁.getKey k mem
+    := by
```

Comment:
Our style guide wants this indentation:
```suggestion
    (∃ (mem₂ : m₂.contains k), (m₁.union m₂).getKey k (contains_union_of_left h₁ h₂ mem) = m₂.getKey k mem₂) ∨ 
      (m₁.union m₂).getKey k (contains_union_of_left h₁ h₂ mem) = m₁.getKey k mem := by
```
Note that the `∨` is at the end of the line, not at the beginning, and similar for the `:= by`.